evalml/tests/data_checks_tests/test_datetime_nan_data_check.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import pandas as pd
+import woodwork as ww
+
+from evalml.data_checks import (
+    DataCheckError,
+    DataCheckMessageCode,
+    DateTimeNaNDataCheck
+)
+
+
+def test_highly_null_data_check_errors(ts_data):
+    data, _ = ts_data
+    data.reset_index(inplace=True, drop=False)
+    data.at[0, 'index'] = np.NaN
+    dt_nan_check = DateTimeNaNDataCheck()
+    assert dt_nan_check.validate(data) == {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input datetime column (index) contains NaN values. Please input NaN values or drop this column.',
+                                  data_check_name=DateTimeNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
+                                  details={"column": 'index'})]
+    }
+
+
+def test_highly_null_data_check_input_formats():
+    dt_nan_check = DateTimeNaNDataCheck()
+
+    # test empty pd.DataFrame
+    assert dt_nan_check.validate(pd.DataFrame()) == {"warnings": [], "errors": [], "actions": []}
+
+    expected = {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input datetime column (index) contains NaN values. Please input NaN values or drop this column.',
+                                  data_check_name=DateTimeNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
+                                  details={"column": 'index'})]
+    }
+
+    dates = np.arange(np.datetime64('2017-01-01'), np.datetime64('2017-01-08'))
+    dates[0] = np.datetime64('NaT')
+
+    #  test Woodwork
+    ww_input = ww.DataTable(pd.DataFrame(dates, columns=['index']))
+    assert dt_nan_check.validate(ww_input) == expected
+
+    expected = {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input datetime column (0) contains NaN values. Please input NaN values or drop this column.',
+                                  data_check_name=DateTimeNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
+                                  details={'column': 0})]
+    }
+
+    #  test 2D list
+    assert dt_nan_check.validate([dates, np.arange(np.datetime64('2017-01-01'), np.datetime64('2017-01-08'))]) == expected
+
+    # test np.array
+    assert dt_nan_check.validate(np.array([dates, np.arange(np.datetime64('2017-01-01'), np.datetime64('2017-01-08'))])) == expected
```

Comment:
Can we add test coverage for 
- Dataframe with other columns (with null values), no datetimes
- Dataframe with multiple datetime columns, >=2 columns with nulls
- Dataframe with multiple datetime columns, 0 columns with null
- Dataframe with datetime column with 1 value as empty string: ''

Just want to make sure the code works as we'd expect in these scenarios

Reply:
Got it. I can add all the tests except maybe the last one; I don't think I can mix types (it'll complain that there is a string if I set the column to datetime) in a pandas DF but I'll give it a go